### PR TITLE
Adding and fixing some tests

### PR DESCRIPTION
**Pull request by alanking**
Addresses #100 
Addresses #280 
Addresses #288 

All tests are passing.

### CODE REVIEW

**Comment by alanking** [comment]
Added a bail-out for use of DeleteMode with S3 bucket syncs. #282 will add actual support.

File: irods_capability_automated_ingest/test/test_delete_modes.py [hunk 1]
```diff
@@ -592,6 +577,56 @@ def test_PUT_APPEND_and_NO_TRASH_deletes_collections(self):
...
+                self.assertFalse(
+                    self.irods_session.data_objects.exists(
+                        "/".join([self.target_subcollection_for_removal, file])
+                    ),
+                )
+            # The collection should still exist.
+            self.assertTrue(
+                self.irods_session.collections.exists(
+                    self.target_subcollection_for_removal
+                ),
```
Comment (korydraughn) [inline irods_capability_automated_ingest/test/test_delete_modes.py:627]: So the black formatter is adding commas to the end of functions too?
Does `assertTrue` accept a second parameter (i.e. a message or something)?

**Review by korydraughn — APPROVED** [review]
Changes look good.

File: irods_capability_automated_ingest/test/test_delete_modes.py (hunk 1 above)
Comment (alanking) [reply irods_capability_automated_ingest/test/test_delete_modes.py:627]: Yes to both. Black formatter is doing most of these line separations and comma-additions. `assertTrue` accepts an optional `msg` parameter: https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertTrue

**Comment by alanking** [comment]
#'d, mergin.

**Pull request merged**